src/loaders/ttf/tvgTtfReader.cpp

```diff
@@ -389,10 +389,11 @@ uint32_t TtfReader::glyph(uint32_t codepoint, TtfGlyphMetrics& gmetrics)
     auto glyph = this->glyph(codepoint);
     if (glyph == INVALID_GLYPH) {
         TVGERR("TTF", "invalid glyph id, codepoint(0x%x)", codepoint);
-        return INVALID_GLYPH;
+    } else if(!glyphMetrics(glyph, gmetrics)) {
```

Comment:
Considering your first commit, it looks you don't need to change the code in glyph().
Could you please squash commits into one to make it clearly visible?